File: 165_compare_version_numbers/compareVersion.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        l1, l2 = version1.split("."), version2.split("."),
        n1, n2 = len(l1),len(l2)
        n = max(n1, n2)

        for i in range(n):
            v1 = int(l1[i]) if i < n1 else 0
            v2 = int(l2[i]) if i < n2 else 0
            if v1 > v2:
                return 1
            elif v1 < v2:
                return -1

        return 0

File: 165_compare_version_numbers/compareVersion.java
class Solution {
    public int compareVersion(String version1, String version2) {
        String[] l1 = version1.split("\\.");
        String[] l2 = version2.split("\\.");

        int n1 = l1.length;
        int n2 = l2.length;

        int n = Math.max(n1, n2);

        for (int i = 0; i < n; i++) {
            Integer v1 = i < n1 ? Integer.parseInt(l1[i]) : 0;
            Integer v2 = i < n2 ? Integer.parseInt(l2[i]) : 0;

            int v = v1.compareTo(v2);
            if (v != 0) {
                return v;
            }
        }
        return 0;
    }
}